7_react/react1/week3/src/components/TodoList.jsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import "./TodoList.css";
+
+const BorderComponent = ({ children }) => {
+  return (
+    <div style={{ border: "2px solid black", padding: "10px" }}>{children}</div>
+  );
+};
+
+BorderComponent.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+const TodoList = () => {
+  const [todos, setTodos] = useState([]);
+  const [timer, setTimer] = useState(0);
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [editTodoId, setEditTodoId] = useState(null);
+  const [editedDescription, setEditedDescription] = useState("");
+  const [editedDeadline, setEditedDeadline] = useState("");
+
+  useEffect(() => {
+    fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    )
+      .then((response) => response.json())
+      .then((data) => setTodos(data))
+      .catch((error) => console.error("Error fetching todos:", error));
+
+    const interval = setInterval(() => {
+      setTimer((prevTimer) => prevTimer + 1);
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    if (name === "description") setDescription(value);
+    if (name === "deadline") setDeadline(value);
+    if (name === "editedDescription") setEditedDescription(value);
+    if (name === "editedDeadline") setEditedDeadline(value);
+  };
+
+  const addTodo = () => {
+    if (!description || !deadline || new Date(deadline) < new Date()) return;
+    const newTodo = {
+      id: Date.now(),
+      description: description,
+      deadline: deadline,
+      done: false,
+    };
+    setTodos((prevTodos) => [...prevTodos, newTodo]);
+    setDescription("");
+    setDeadline("");
+  };
+
+  const toggleDone = (id) => {
+    setTodos((prevTodos) =>
+      prevTodos.map((todo) =>
+        todo.id === id ? { ...todo, done: !todo.done } : todo
+      )
+    );
+  };
+
+  const deleteTodo = (id) => {
+    setTodos((prevTodos) => prevTodos.filter((todo) => todo.id !== id));
+  };
+
+  const editTodo = (id, description, deadline) => {
+    setEditTodoId(id);
+    setEditedDescription(description);
+    setEditedDeadline(deadline);
+  };
+
+  const updateTodo = () => {
+    if (
+      !editedDescription ||
+      !editedDeadline ||
+      new Date(editedDeadline) < new Date()
```

Comment:
I would suggest to throw an error  with text ' Pay attention to time' or similar 😄  if I am trying to edit todo and setting the time in the past, for example yesterday. It is a bit confusing in this way.

Reply:
okay, will definitely take this into consideration.